pkg/controller/keyvault/kv_util.go

```diff
@@ -127,23 +127,23 @@ func certSecretName(ingressName string) string {
 	return fmt.Sprintf("keyvault-%s", ingressName)
 }
 
-type userError struct {
+type UserError struct {
```

Comment:
don't really like exporting this from kv package. weird for this to be under kv package really. 

Move to utils where it makes sense? Or maybe a new errors pkg?

Reply:
if we export from kv it's just kinda weird and the more we do that the more chance of circular dependency

Reply:
resolving after refactor PR was merged in